openwisp_controller/config/tests/test_device.py

```diff
@@ -389,8 +389,10 @@ def test_device_group_changed_emitted(self):
             )
 
     def test_device_group_changed_not_emitted_on_creation(self):
+        org = self._get_org()
+        device_group = self._create_device_group(organization=org)
         with catch_signal(device_group_changed) as handler:
-            self._create_device(organization=self._get_org())
+            self._create_device(name='test', organization=org, group=device_group)
```

Comment:
I have updated this test to include the scenario shared by @codesankalp in https://github.com/openwisp/openwisp-controller/issues/631#issuecomment-1127532930

> I checked that `device_group_changed` gets fired on creation of device also if device group is selected while creating the device. So, I think we don't need to implement post_save signal for device model.